cmake/CxxVsF90.cmake.in

```diff
@@ -0,0 +1,41 @@
+# Run cprnc on each of the output files against the target
+# Note: we mix @ and $ notation, since at configure time we only
+#       resolve @ variables. The remaning ones will be resolve by
+#       cmake when the script is run with 'cmake -P CprncCxxVsF90.cmake'
+FOREACH (NC_FILE_NAME @NC_OUTPUT_FILES@)
+  EXECUTE_PROCESS (COMMAND  @CPRNC_BINARY@ @F90_DIR@/movies/${NC_FILE_NAME} @CXX_DIR@/movies/${NC_FILE_NAME}
+                   RESULT_VARIABLE ERROR_CODE
+                   OUTPUT_VARIABLE cprnc_stdout
+                   ERROR_VARIABLE cprnc_stderr)
+  MESSAGE (${cprnc_stdout})
+
+  # Search stdout for "IDENTICAL". -1 means it does not exist. Use REVERSE on
+  # the off chance that makes it faster, since "IDENTICAL", if it exists, is
+  # near the end of the output.
+  string (FIND "${cprnc_stdout}" "IDENTICAL" identical_pos REVERSE)
+
+  # If cprnc does not return 0, then the test failed
+  IF (identical_pos EQUAL -1)
+    MESSAGE (FATAL_ERROR "Test did not succeed. Netcdf outputs differ.")
+  ENDIF()
+ENDFOREACH()
+
+# Grep F90 and CXX raw output for all lines containing the string '/dt'.
+# As of today, these are all and only the lines where diagnostics are printed.
+# Then compare the output strings. If they are equal, then the two tests print
+# the same diagnostics
+
+EXECUTE_PROCESS (COMMAND grep /dt @F90_DIR@/@F90_TEST_NAME@_1.out
```

Comment:
We could also compare output for lines
    prim_state_mod.F90:356-379
as part of this check. (These lines are part of the diagnostic
output, although they are not part of the new diagnostic code in
this PR.) IMO, there is nothing wrong (and I find in general it
is useful for all scripting) with adding a greppable line prefix
to all diagnostic lines, something like "diag>". Then you'd grep
for "^diag>" and get all desired lines rather than "/dt".

Reply:
Sounds reasonable. I can do that.